### ISSUE TITLE
Enhancing the `test` library or adding scripts to `examples`

### ISSUE THREAD

**Issue by ronaldtse** [riboseinc/retrace#205]
Everyone seems to have their own little test utility for each function that is not in `test`, like these:

* https://github.com/riboseinc/retrace/pull/196#issuecomment-313281866
* https://github.com/riboseinc/retrace/pull/195#issuecomment-313404067

Maybe they should be incorporated into `tests` or put in a separate `examples` directory?

**Comment by jjr840430** [comment]
I think 'examples' directory is desired.

Everyone may create a directory, and add sources, README and the configuration file for each own tests. 
For example, we may add practical examples such as rnp or valgrind leak testing there.
BTW, it doesn't need to be added into automake building.

I think current 'tests' directory is just for testing single operation of syscalls.

**Comment by ronaldtse** [comment]
I agree we need an examples directory. However, since the test directory mainly focuses on single function call tests, maybe we can incorporate some of these more comprehensive ones, such as for getenv fuzzing, into an integration test suite so we test the full functionality (the features).

**Comment by erikbor** [comment]
I just merged [#220](https://github.com/riboseinc/retrace/pull/220) which has an `examples` directory with a getenv-fuzzing directory containing a C file that contains two vulnerabilities which can be triggered using retrace getenv fuzzing. A README.md and retrace configuration files have been provided. I think this is a good approach, let me know if you think this approach needs to be changed and/or formalised (somewhere).

**Comment by erikbor** [comment]
I also added examples for getuid() redirection: [#233](https://github.com/riboseinc/retrace/pull/223)
